src/NServiceBus.Core.Tests/Faults/MoveFaultsToErrorQueueTests.cs

```diff
@@ -12,89 +12,145 @@ namespace NServiceBus.Core.Tests
     [TestFixture]
     public class MoveFaultsToErrorQueueTests
     {
-        [Test]
-        public async Task ShouldForwardToErrorQueueForAllExceptions()
+        FakeCriticalError criticalError;
+        FakeFaultPipeline faultPipeline;
+
+        [SetUp]
+        public void Setup()
         {
-            var fakeFaultPipeline = new FakeFaultPipeline();
-            var errorQueueAddress = "error";
-            var behavior = new MoveFaultsToErrorQueueBehavior(
-                new FakeCriticalError(),
-                errorQueueAddress,
-                "public-receive-address");
+            faultPipeline = new FakeFaultPipeline();
+            criticalError = new FakeCriticalError();
+        }
 
-            var context = CreateContext("someid", fakeFaultPipeline);
+        [Test]
+        [TestCase(TransportTransactionMode.None)]
+        [TestCase(TransportTransactionMode.ReceiveOnly)]
+        [TestCase(TransportTransactionMode.SendsAtomicWithReceive)]
+        [TestCase(TransportTransactionMode.TransactionScope)]
+        public async Task ShouldForwardToErrorQueueForAllExceptions(TransportTransactionMode transactionMode)
+        {
+            var behavior = CreateBehavior(transactionMode, "errors-queue");
+            var context = CreateContext("message-id");
 
-            await behavior.Invoke(context, () => { throw new Exception("testex"); });
+            await behavior.Invoke(context, () => { throw new Exception(); });
 
-            Assert.AreEqual(errorQueueAddress, fakeFaultPipeline.Destination);
+            if (transactionMode != TransportTransactionMode.None)
+            {
+                await behavior.Invoke(context, () => Task.FromResult(0));
+            }
 
-            Assert.AreEqual("someid", fakeFaultPipeline.MessageSent.MessageId);
+            Assert.AreEqual("errors-queue", faultPipeline.Destination);
+            Assert.AreEqual("message-id", faultPipeline.MessageSent.MessageId);
         }
 
         [Test]
```

Comment:
This is redundant. `[TestCase]` is enough for NUnit to run this as a test method.